pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -326,6 +331,14 @@ func (s *NodeController) checkNodeReady(node *api.Node) *api.NodeCondition {
 	}
 }
 
+// sendConditionEvent sends condition change event.
+func (s *NodeController) sendConditionEvent(node *api.Node, kind api.NodeConditionKind, oldCondition, newCondition *api.NodeCondition) {
+	// TODO: if 'oldCondition' is nil, we should publish the initial condition, but selfLink is not set.
+	if oldCondition != nil {
```

Comment:
Yeah, we should. But if it had no old condition, then it is first checked in node controller, which means it hasn't been created yet.  So we failed to send event because of missing self-link, which is added by apiserver I think.  I haven't looked into it deeply, but plan to take a look later, hence the TODO.  Any quick idea?